module3/homework/vectorOfSharedPointers/vectorFunctions.cpp

```diff
@@ -0,0 +1,32 @@
+#include "vectorFunctions.hpp"
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> generate(int count){
+    std::vector<std::shared_ptr<int>> newVector;
+    newVector.reserve(count);
+    for (size_t i = 0 ; i < count ; ++i) {
+        newVector.emplace_back(std::make_shared<int>(i));
+    }
+    return newVector;
+}
+void print(std::vector<std::shared_ptr<int>>& newVector){
+    for (const auto& element : newVector){
+        std::cout << *element << "\n";
+  }
+}
+void add10(std::vector<std::shared_ptr<int>>& newVector){
+  for (const auto& element : newVector){
+    if (element != nullptr)
+     *element += 10;
```

Comment:
I would add { } block for if statement, even if it has only one line of code inside. The same as you have in `void sub10(int* const ptr)` function